src/header/internals/item.rs

```diff
@@ -82,6 +82,14 @@ impl Item {
         }
         self.typed.get_mut(tid).map(|typed| unsafe { typed.downcast_mut_unchecked() })
     }
+
+    pub fn into_typed<H: Header>(self) -> Option<H> {
```

Comment:
To be clear, I get a type conflict exactly at this spot: Expected `H`, got `Box<Header + Send + Sync>`. The first idea here is to add a new `downcast_*` method to `impl Header + Send + Sync` (again analogous to the `get` methods), but I don't know how to implement such a method.